compiler/src/iree/compiler/Dialect/Flow/Transforms/VerifyInputLegality.cpp

```diff
@@ -17,12 +17,40 @@ namespace iree_compiler {
 namespace IREE {
 namespace Flow {
 
+static llvm::cl::opt<bool> clVerifyNoF64(
+    "iree-flow-verify-no-f64",
+    llvm::cl::desc("Verify that there are no ops that use f64."),
+    llvm::cl::init(true));
+
+// TODO: This helper function should be in MLIR.
+static Type getTensorOrVectorElementType(Type type) {
+  if (auto vec = llvm::dyn_cast<VectorType>(type))
+    return vec.getElementType();
+
+  // Look through tensor<vector<...>> to find the underlying element type.
+  if (auto tensor = llvm::dyn_cast<TensorType>(type))
+    return getTensorOrVectorElementType(tensor.getElementType());
+  return type;
+}
+
 namespace {
+
 class VerifyInputLegalityPass
     : public VerifyInputLegalityBase<VerifyInputLegalityPass> {
   void runOnOperation() override {
     ConversionTarget target(getContext());
-    target.markUnknownOpDynamicallyLegal([](Operation *) { return true; });
+    target.markUnknownOpDynamicallyLegal([](Operation *op) {
+      if (!clVerifyNoF64)
+        return true;
+      bool usesFloat64 = llvm::any_of(
+          llvm::concat<Type>(op->getOperandTypes(), op->getResultTypes()),
+          [](Type type) {
+            return llvm::isa<Float64Type>(getTensorOrVectorElementType(type));
+          });
+      if (usesFloat64)
+        op->emitError() << "uses unsupported type float64";
```

Comment:
```suggestion
        op->emitError() << "uses partially supported type f64; pass --iree-flow-verify-no-f64=false to allow anyway";
```

We want to support f64 everywhere and some backends already do so as this change now makes using it impossible by default we should make it easy for people still wanting to allow it through.

Reply:
done